File: control-center/scripts/split_tabs.py
import re

with open("src/App.tsx", "r", encoding="utf-8") as f:
    content = f.read()

# Find the main return block
start_idx = content.find("return (")
if start_idx == -1:
    print("Could not find return block")
    exit(1)

body = content[start_idx:]

# The tabs are structured like: `) : activeTab === "xyz" ? (` or `{activeTab === "xyz" ? (`
tabs = [
    "dashboard", "global", "dm", "kanban", "runs", "skills", 
    "channels", "files", "missions", "agents", "models", "system", "org_dashboard"
]

import os
os.makedirs("src/components/tabs", exist_ok=True)

for tab in tabs:
    # A very naive search to find the start of the tab's JSX
    # This might fail on nested ternary, but let's see.
    pattern = rf'activeTab === "{tab}"(\s*&&\s*activeOrgNode)?\s*\?\s*\('
    match = re.search(pattern, body)
    if match:
        print(f"Found start of {tab}")
    else:
        print(f"Could not find {tab}")

File: tools/legacy-repair/split_tabs.js
const fs = require('fs');

const content = fs.readFileSync("src/App.tsx", "utf8");
const returnStart = content.indexOf('  return (\n    <div className="app-layout">');

if (returnStart === -1) {
  console.log("Could not find return start");
  process.exit(1);
}

const header = content.substring(0, returnStart);
const jsx = content.substring(returnStart);

// We will split the JSX by searching for `activeTab === "`
const tabs = [
  "dashboard", "global", "dm", "kanban", "runs", "skills", 
  "channels", "files", "missions", "agents", "models", "system", "org_dashboard"
];

for (const tab of tabs) {
  const marker = `activeTab === "${tab}"`;
  const idx = jsx.indexOf(marker);
  if (idx !== -1) {
    console.log(`Found tab: ${tab} at index ${idx}`);
  } else {
    console.log(`Could not find tab: ${tab}`);
  }
}
